ucdavis_service/src/main/java/com/ucdavis/application/UCDavisService.java

```diff
@@ -140,6 +140,47 @@ public void writeToMap(List<JSONObject> records, String target) {
 		} else {
 			data.put(target, dataItems);
 		}
+		
+		checkNullKeysAndValue(dataItems, true);
+	}
+	
+	private void checkNullKeysAndValue(Map<String, Map<String,String>> dataItems, boolean isDaily) {
+		// daily keys
+		String[] dailyKeys = {"Date","DayAirTmpMax","DayWindRun","DayAsceEto","DaySolRadAvg","DayWindSpdAvg","DayPrecip","Standard","DayVapPresAvg","DaySoilTmpAvg","Station","DayRelHumMax","DayAirTmpAvg","DayAirTmpMin","Scope","DayDewPnt","DayRelHumAvg","DayRelHumMin","Julian","ZipCodes"};
+		
+		// TODO hourly keys
+		String[] hourlyKeys = {};
+		
+		SortedSet<String> sortedDateKeys = new TreeSet<String>(dataItems.get("Date").keySet()); // dates cant be null
```

Comment:
How you check the null date?

Reply:
For CIMIS service, the date won't be null, that's why I put the key "Date" search here. So should I assume the case exists that dates could be null?